src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -0,0 +1,129 @@
+using System.IO;
+using UnityEditor.iOS.Xcode;
+using UnityEditor.iOS.Xcode.Extensions;
+using UnityEngine;
+
+namespace Sentry.Unity.Editor.iOS
+{
+    internal class SentryXcodeProject
+    {
+        private const string FrameworkName = "Sentry.framework";
+
+        private const string MainPath = "MainApp/main.mm";
+        private const string OptionsPath = "MainApp/SentryOptions.m";
+
+        private readonly string _pathToProject;
+        private readonly PBXProject _project;
+        private readonly string _projectPath;
+
+        private readonly INativeMain _nativeMain;
+        private readonly ISentryNativeOptions _sentryNativeOptions;
+
+        private readonly string _relativeFrameworkPath;
+
+        internal SentryXcodeProject(
+            string pathToProject,
+            INativeMain? mainModifier = null,
+            ISentryNativeOptions? sentryNativeOptions = null)
+        {
+            _pathToProject = pathToProject;
+
+            _projectPath = PBXProject.GetPBXProjectPath(pathToProject);
+            _project = new PBXProject();
+
+            if (File.Exists(_projectPath))
+            {
+                _project.ReadFromString(File.ReadAllText(_projectPath));
+            }
+            else
+            {
+                Debug.LogWarning($"Could not locate generated Xcode project at {_projectPath}");
+            }
+
+            _relativeFrameworkPath = GetRelativeFrameworkPath(_pathToProject);
+
+            _nativeMain = mainModifier ?? new NativeMain();
+            _sentryNativeOptions = sentryNativeOptions ?? new NativeOptions();
+        }
+
+        public static SentryXcodeProject Open(string path)
+        {
+            return new (path);
```

Comment:
Hm, I thought the formatting for this was:

```suggestion
            return new(path);
```

Did Rider's 'auto format file' turned out as that?

Reply:
Yes! But maybe I got some wrong auto formatting settings?

Reply:
Nope, don't know how I got the space in between there.